# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/VRManager.cpp
# Commit: 898ed02804fe
# Full Hash: 898ed02804fefc9786112f135b9621cc444e6860
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-23 15:52:15
# Regressor Bug: 1550422
# File Overlap Count: 1
# Description:
#   Bug 1550422 - P15. Move Skip and Once gfxPrefs to StaticPrefs. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31259
# ==============================================================================

diff -r e1b7abc99ae9 -r 898ed02804fe gfx/vr/VRManager.cpp
--- a/gfx/vr/VRManager.cpp	Wed May 22 12:44:04 2019 +0000
+++ b/gfx/vr/VRManager.cpp	Wed May 22 22:27:37 2019 +0000
@@ -13,9 +13,9 @@
 #include "mozilla/dom/GamepadEventTypes.h"
 #include "mozilla/layers/TextureHost.h"
 #include "mozilla/layers/CompositorThread.h"
+#include "mozilla/StaticPrefs.h"
 #include "mozilla/Unused.h"
 
-#include "gfxPrefs.h"
 #include "gfxVR.h"
 #include "gfxVRExternal.h"
 
@@ -79,7 +79,7 @@
 #if !defined(MOZ_WIDGET_ANDROID)
   // The VR Service accesses all hardware from a separate process
   // and replaces the other VRSystemManager when enabled.
-  if (!gfxPrefs::VRProcessEnabled() || !XRE_IsGPUProcess()) {
+  if (!StaticPrefs::VRProcessEnabled() || !XRE_IsGPUProcess()) {
     VRServiceManager::Get().CreateService();
   }
   if (VRServiceManager::Get().IsServiceValid()) {
@@ -100,7 +100,7 @@
 
   // Enable gamepad extensions while VR is enabled.
   // Preference only can be set at the Parent process.
-  if (XRE_IsParentProcess() && gfxPrefs::VREnabled()) {
+  if (XRE_IsParentProcess() && StaticPrefs::VREnabled()) {
     Preferences::SetBool("dom.gamepad.extensions.enabled", true);
   }
 }
@@ -144,7 +144,8 @@
   }
   // XRE_IsGPUProcess() is helping us to check some platforms like
   // Win 7 try which are not using GPU process but VR process is enabled.
-  if (XRE_IsGPUProcess() && gfxPrefs::VRProcessEnabled() && mVRServiceStarted) {
+  if (XRE_IsGPUProcess() && StaticPrefs::VRProcessEnabled() &&
+      mVRServiceStarted) {
     RefPtr<Runnable> task = NS_NewRunnableFunction(
         "VRServiceManager::ShutdownVRProcess",
         []() -> void { VRServiceManager::Get().ShutdownVRProcess(); });
@@ -455,7 +456,7 @@
    */
 #if !defined(MOZ_WIDGET_ANDROID)
   if (!mVRServiceStarted) {
-    if (XRE_IsGPUProcess() && gfxPrefs::VRProcessEnabled()) {
+    if (XRE_IsGPUProcess() && StaticPrefs::VRProcessEnabled()) {
       VRServiceManager::Get().CreateVRProcess();
       mVRServiceStarted = true;
     } else {